lib/base/type.hpp

```diff
@@ -83,6 +83,22 @@ class Type : public Object
 	static Type::Ptr GetByName(const String& name);
 	static std::vector<Type::Ptr> GetAllTypes();
 
+	/**
+	 * Returns a list of config types sorted by their "load_after" dependencies.
+	 *
+	 * All dependencies of a given type are listed at a lower index than that of the type itself. In other words,
+	 * if a `Service` type load depends on the `Host` and `ApiListener` types, the Host and ApiListener types are
+	 * guaranteed to appear first on the list. Nevertheless, the order of the Host and ApiListener types themselves
+	 * is arbitrary if the two types are not dependent.
+	 *
+	 * It should be noted that this method does not detect any cyclic load dependencies, instead, it relies on
+	 * the "sort_by_load_after" unit test to fail. Additionally, this will fail fatally when used prior to the
+	 * completion of namespace initialization.
```

Comment:
I wouldn't specify this here. It describes behavior that would mean that the function is broken and there's a test to check that the function isn't broken. So that shouldn't be relevant to any caller. A similar comment is also in the implementation (well, not of this function, but in the function that generates the result for this function). If you added this here because that might help with a failing test, I'd say a comment right above the assertion that would then fail would be more helpful.

Reply:
Note: the selected context is a bit too big. I **don't** want to say this for the following:

> Additionally, this will fail fatally when used prior to the completion of namespace initialization.

It's just regarding cycles. The caller doesn't provide any input that has to be free of cycles. If there were dependency cycles, this function would be broken and there's a tests to check that the function is broken.

To me, that part of the doc comment reads similar to "Note that this function wouldn't work if the implementation of std::sort() was broken"